### PR TITLE
Refactor mocks and clients

### PR DESCRIPTION
**Pull request by kishansagathiya**
Delete `mock` and `clients` package and move client implentations and
mock implementations to respective new packages such as `idler`,
`tenant`, `wit` etc

Fixes #4570

### CODE REVIEW

File: cmd/fabric8-jenkins-proxy/main.go [hunk 1]
```diff
@@ -84,13 +86,13 @@ func main() {
...
-	tenant := clients.NewTenant(config.GetTenantURL(), config.GetAuthToken())
+	tenant := tenant.NewTenant(config.GetTenantURL(), config.GetAuthToken())
 
 	// Create WorkItemTracker client
-	wit := clients.NewWIT(config.GetWitURL(), config.GetAuthToken())
+	wit := wit.New(config.GetWitURL(), config.GetAuthToken())
 
 	// Create Idler client
-	idler := clients.NewIdler(config.GetIdlerURL())
+	idler := idler.New(config.GetIdlerURL())
```
Comment (hrishin) [inline cmd/fabric8-jenkins-proxy/main.go]: :+1: for just `new`

File: internal/storage/mock_store.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package storage
+
+// MockStore is a mock implementation of Store struct.
+// This implementation is meant to be used for testing
+type MockStore struct {
```
Comment (hrishin) [inline internal/storage/mock_store.go]: Its better `MockStore` -> `Mock` for the sake of consistency with other mocks?

**Comment by kishansagathiya** [comment]
[test]

**Comment by kishansagathiya** [comment]
[ test ]

File: internal/storage/store_test.go [hunk 3]
```diff
@@ -13,50 +12,50 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-// TODO turn this into a general in memory store service (HF)
-type mockStore struct {
-	buffer bytes.Buffer
-}
+// // TODO turn this into a general in memory store service (HF)
```
Comment (sthaha) [inline internal/storage/store_test.go]: @kishansagathiya Did you forget to change this part?

Comment (hrishin) [reply internal/storage/store_test.go]: @kishansagathiya are we dropping the implementation?

Comment (kishansagathiya) [reply internal/storage/store_test.go]: It is moved to mock_store.go and exported

**Pull request merged**